src/include/sof/perfcount.h

```diff
@@ -0,0 +1,201 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>
+ */
+
+#ifndef __PERFCOUNT_H__
+#define __PERFCOUNT_H__
+
+#include <stdint.h>
+#include <sof/alloc.h>
+#include <sof/clk.h>
+#include <sof/platform.h>
+#include <sof/list.h>
+#include <sof/drivers/timer.h>
+
+#define PERFCOUNT_TRACE_FMT "avg perf: %u us, %u cycles"
+
+#define PERFCOUNT_NUM_STEPS (1 << CONFIG_PERFCOUNT_HISTORY_LOG2)
+#define PERFCOUNT_AVG_SHIFT CONFIG_PERFCOUNT_HISTORY_LOG2
+
+/* TODO: Enable traces after runtime trace configuration changes */
+#if 0
+#define _perfcount_trace(context, prefix) \
```

Comment:
Please align all trailing backslashes in a column, for better readability of multiline macro definitions, similarly to trace_event macros family in trace.h.